### PR TITLE
update jenkins deployment to use docker on agent nodes

### PR DESCRIPTION
**Pull request by abhinavGupta16**
Signed-off-by: Abhinav Gupta <abhng@amazon.com>

### Description
ci can now run docker agent nodes and withAws commands
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-ci/issues/79
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: lib/compute/agent-nodes.ts [hunk 1]
```diff
@@ -35,7 +35,8 @@ export class AgentNode {
 
   public readonly InitScript: string = 'sudo mkdir -p /var/jenkins/ && sudo chown -R ec2-user:ec2-user /var/jenkins '
     + '&& sudo yum install -y java-1.8.0-openjdk cmake python3 python3-pip && sudo yum groupinstall -y \'Development Tools\' '
-    + '&& sudo ln -sfn `which pip3` /usr/bin/pip && pip3 install pipenv && sudo ln -s ~/.local/bin/pipenv /usr/local/bin';
+    + '&& sudo ln -sfn `which pip3` /usr/bin/pip && pip3 install pipenv && sudo ln -s ~/.local/bin/pipenv /usr/local/bin '
```
Comment (gaiksaya) [inline lib/compute/agent-nodes.ts:38]: Would it make sense to move this init script at config level so that it is separate for each agent node?
We started with common script because both were AL2 agent nodes but Ubuntu, windows, etc will be different. Can be part of another PR though.

**Review by gaiksaya — APPROVED** [review]

File: lib/compute/jenkins-plugins.ts [hunk 2]
```diff
@@ -92,6 +92,7 @@ export class JenkinsPlugins {
       'ownership',
       'pam-auth',
       'parameterized-scheduler',
+      'pipeline-aws',
```
Comment (peterzhuamazon) [inline lib/compute/jenkins-plugins.ts:95]: Why is this needed to be added? What purpose?

Comment (peterzhuamazon) [reply lib/compute/jenkins-plugins.ts:95]: withAWS?

File: lib/compute/agent-nodes.ts (hunk 1 above)
Comment (peterzhuamazon) [reply lib/compute/agent-nodes.ts:38]: We havent think about windows for now, but only using AL2 for LINUX Agent is enough, as AL2 can provision Ubuntu container for buildx purposes.

Comment (gaiksaya) [reply lib/compute/agent-nodes.ts:38]: `AL2 can provision Ubuntu container for buildx purposes` I didn't get that part? You mean we can pull ubuntu image on AL2?

Comment (abhinavGupta16) [reply lib/compute/agent-nodes.ts:38]: I tried that and it works

Comment (abhinavGupta16) [reply lib/compute/agent-nodes.ts:38]: I was able to pull the ubuntu docker image on AL2X64 and run the job

**Review by zelinh — APPROVED** [review]
